### PR TITLE
Add sign out button

### LINKED ISSUE #305: Add sign out option in the header
Currently, user cannot not sign out or switch account after he, she or they signed in.

### PR DESCRIPTION
**Pull request by kapillamba4**
### Description
Added sign out button to header
### Screenshots
Screenshot from 2020-03-03 22-44-45

### CODE REVIEW

**Comment by kapillamba4** [comment]
@byliuyang

File: frontend/src/component/ui/Button.tsx [hunk 1]
```diff
@@ -15,6 +15,6 @@ export class Button extends Component<Props> {
   };
 
   render() {
-    return <button onClick={this.handleClick}>{this.props.children}</button>;
+    return <button className="btn" onClick={this.handleClick}>{this.props.children}</button>;
```
Comment (byliuyang) [inline frontend/src/component/ui/Button.tsx]:
```suggestion
    return <button onClick={this.handleClick}>{this.props.children}</button>;
```

**Comment by byliuyang** [comment]
Hi @kapillamba4, thank you so much for working on this feature. I left few comments on the PR.

File: frontend/src/component/pages/shared/Header.scss [hunk 2]
```diff
@@ -21,4 +21,11 @@ header {
     letter-spacing: 2px;
     line-height: 30px;
   }
+
+  .sign-out {
+    margin-left: auto;
+    button {
+      height: 100%;
+    }
```
Comment (byliuyang) [inline frontend/src/component/pages/shared/Header.scss]: Let's open an issue and add `// TODO(issue#your_issue_number): remove dependency on internal implementation of the Button.` before those lines.

File: frontend/src/component/pages/shared/Header.tsx [hunk 3]
```diff
@@ -1,12 +1,20 @@
...
-export class Header extends Component {
+interface Props {
+  onSignOutButtonClick: () => void;
+}
+
+export class Header extends Component<Props> {
   render() {
     return (
       <header>
         <div className={'center'}>
```
Comment (byliuyang) [inline frontend/src/component/pages/shared/Header.tsx]: Let's open an issue and add `<!-- TODO(issue#your_issue_number): use flex layout to prepare for search box with dynamic width. -->` before those lines.

You can read about flex layout here!!!: https://css-tricks.com/snippets/css/a-guide-to-flexbox

Comment (byliuyang) [reply frontend/src/component/pages/shared/Header.tsx]: Discussed offline. This is not needed anymore.

File: frontend/src/component/pages/shared/Header.tsx [hunk 4]
```diff
@@ -1,12 +1,20 @@
...
           <div id="logo">Short</div>
+          <div className={'sign-out'}>
+            <Button onClick={this.props.onSignOutButtonClick}>Sign out</Button>
+          </div>
```
Comment (byliuyang) [inline frontend/src/component/pages/shared/Header.tsx]:
```suggestion
        <div className={'center'}>
          <div id="logo">Short</div>
          <div id="spacer">// This is where we are going to add search box. We call it spacer for now.</div>
          <div className={'sign-out'}>
            <Button onClick={this.props.onSignOutButtonClick}>Sign out</Button>
          </div>
```

```css
.spacer {
    flex: 1;
}
```
Then you remove `margin-left: auto;`

**Pull request merged**